homeassistant/components/kostal_plenticore/sensor.py

```diff
@@ -0,0 +1,192 @@
+from typing import Any, Callable, Dict, Optional
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_DEVICE_CLASS, ATTR_ICON, ATTR_UNIT_OF_MEASUREMENT
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    ATTR_ENABLED_DEFAULT,
+    DOMAIN,
+    SENSOR_PROCESS_DATA,
+    SENSOR_SETTINGS_DATA,
+)
+from .helper import (
+    PlenticoreDataFormatter,
+    ProcessDataUpdateCoordinator,
+    SettingDataUpdateCoordinator,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+):
+    """Add kostal plenticore Sensors."""
+    plenticore = hass.data[DOMAIN][entry.entry_id]
+
+    entities = []
+
+    available_process_data = await plenticore.client.get_process_data()
+    process_data_update_coordinator = ProcessDataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        "Process Data",
+        timedelta(seconds=10),
+        plenticore,
+    )
+    for module_id, data_id, name, sensor_data, fmt in SENSOR_PROCESS_DATA:
+        if (
+            module_id not in available_process_data
+            or data_id not in available_process_data[module_id]
+        ):
+            _LOGGER.debug(
+                "Skipping non existing process data %s/%s", module_id, data_id
+            )
+            continue
+
+        entities.append(
+            PlenticoreDataSensor(
+                process_data_update_coordinator,
+                entry.entry_id,
+                entry.title,
+                module_id,
+                data_id,
+                name,
+                sensor_data,
+                PlenticoreDataFormatter.get_method(fmt),
+                plenticore.device_info,
+            )
+        )
+
+    available_settings_data = await plenticore.client.get_settings()
+    settings_data_update_coordinator = SettingDataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        "Settings Data",
+        timedelta(seconds=300),
+        plenticore,
+    )
+    for module_id, data_id, name, sensor_data, fmt in SENSOR_SETTINGS_DATA:
+        if module_id not in available_settings_data or data_id not in (
+            setting.id for setting in available_settings_data[module_id]
+        ):
+            _LOGGER.debug(
+                "Skipping non existing setting data %s/%s", module_id, data_id
+            )
+            continue
+
+        entities.append(
+            PlenticoreDataSensor(
+                settings_data_update_coordinator,
+                entry.entry_id,
+                entry.title,
+                module_id,
+                data_id,
+                name,
+                sensor_data,
+                PlenticoreDataFormatter.get_method(fmt),
+                plenticore.device_info,
+            )
+        )
+
+    async_add_entities(entities)
+
+
+class PlenticoreDataSensor(CoordinatorEntity):
```

Comment:
Also inherit from `SensorEntity` imported from the sensor integration.
```suggestion
class PlenticoreDataSensor(CoordinatorEntity, SensorEntity):
```